lib/commands/project.ts

```diff
@@ -28,10 +31,125 @@ export class ProjectCommandBase implements ICommand {
 	}
 }
 
+export class InitProjectCommandBase extends ProjectCommandBase {
+	protected projectDir: string;
+	protected tnsModulesDir: string;
+	protected bootstrapFile: string;
+	protected indexHtml: string;
+
+	constructor(protected $project: Project.IProject,
+				protected $errors: IErrors,
+				protected $fs: IFileSystem,
+				protected $logger: ILogger,
+				protected $projectNameValidator?: IProjectNameValidator) {
+		super($project, $errors, $projectNameValidator);
+
+		this.projectDir = this.$project.getNewProjectDir();
+		this.tnsModulesDir = path.join(this.projectDir, "tns_modules");
+		this.bootstrapFile = path.join(this.projectDir, "app", "bootstrap.js");
+		this.indexHtml = path.join(this.projectDir, "index.html");
+	}
+
+	public execute(args: string[]): IFuture<void> {
+		return null;
+	}
+
+	allowedParameters: ICommandParameter[] = [];
+	protected isWebsiteProject(): IFuture<boolean> {
+		return ((): boolean => {
+			var result = true;
+
+			if(!this.$fs.exists(this.indexHtml).wait()) {
+				this.$logger.trace("Missing %s file, so this is not a mobile website project.", this.indexHtml);
+				result = false;
+			}
+
+			// if there's at least one cordova file, we can't say this is a website project
+			Object.keys(MobileHelper.platformCapabilities).forEach((platform) => {
```

Comment:
this somewhat complex for each is written is 3 places, can we extract it to a helper?